File: test/language/templates/python/StructTemplatedTemplateArgumentTest.py
import unittest
import zserio

from testutils import getZserioApi

class StructTemplatedTemplateArgumentTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "templates.zs").struct_templated_template_argument

    def testReadWrite(self):
        structTemplatedTemplateArgument = self.api.StructTemplatedTemplateArgument(
            self.api.Field_Compound_uint32(self.api.Compound_uint32(42))
        )

        writer = zserio.BitStreamWriter()
        structTemplatedTemplateArgument.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readStructTemplatedTemplateArgument = self.api.StructTemplatedTemplateArgument()
        readStructTemplatedTemplateArgument.read(reader)
        self.assertEqual(structTemplatedTemplateArgument, readStructTemplatedTemplateArgument)

File: test/language/templates/cpp/StructTemplatedTemplateArgumentTest.cpp
#include "gtest/gtest.h"

#include "templates/struct_templated_template_argument/StructTemplatedTemplateArgument.h"

namespace templates
{
namespace struct_templated_template_argument
{

TEST(StructTemplatedTemplateArgumentTest, readWrite)
{
    StructTemplatedTemplateArgument structTemplatedTemplateArgument;
    structTemplatedTemplateArgument.setCompoundField(Field_Compound_uint32{Compound_uint32{42}});

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    structTemplatedTemplateArgument.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    StructTemplatedTemplateArgument readStructTemplatedTemplateArgument(reader);

    ASSERT_TRUE(structTemplatedTemplateArgument == readStructTemplatedTemplateArgument);
}

} // namespace struct_templated_template_argument
} // namespace templates
